### ISSUE TITLE
Deadlinks

### ISSUE THREAD

**Issue by gcharest** [sara-sabr/ITStrategy#157]
When using links to documents, they must not move.

URL testing happens in our tests doesn't prevent URLs in emails to be dead in the future.

**Comment by gcharest** [comment]
Need to ensure all URLs are also tested as part of the generated material. There seems to be instances where it isn't in scope of the current CI tests.

**Comment by gcharest** [comment]
Also realized that the URLs generated via the draw.io graphic are never assessed as the local file is a compressed XML: our tests wouldn't be able to test those!

**Comment by remyb2canada** [comment]
those compressed XML are rendered as HTML in the browser though. There must be something to leverage and use to test

**Comment by gcharest** [comment]
Tests are ran before the site is rendered, we need deploy tests before
merging

Le ven. 2 août 2019 08 h 25, Rémy Bernard <notifications@github.com> a
écrit :

**Comment by smellems** [comment]
For ORE we run HTMLProofer and HTML5Validator.  I think they can both check links on the site.

But it wont fix old links we shared before we created the sebsite

https://github.com/canada-ca/ore-ero/blob/master/.travis.yml

**Comment by gcharest** [comment]
Definitely, those are two issues we need to work separately. Part one is
going to be fixed in my next PR (moving all documents into pages by default)

Part two, we'd need the proofing you just mentioned.

Le ven. 2 août 2019 10 h 06, Sébastien Lemay <notifications@github.com> a
écrit :

**Comment by remyb2canada** [comment]
@gcharest can we discuss versioning before the PR?
I.e. how do we communicate the maturity of documents/pages from "an idea" to "work in progress" to "approved"

**Comment by gcharest** [comment]
@remyb2canada moving the files won't display them until we add the front matter so it's essentially the same as leaving them in the wip-tec folder for us.

See Contributing [Jekyll Pages](https://github.com/sara-sabr/ITStrategy/blob/master/CONTRIBUTING.md#jekyll-pages) info for more details :D

**Comment by remyb2canada** [comment]
:+1:

**Comment by gcharest** [comment]
In fact, this is more consistent as if we keep a document in wip-tec but are ready to "publish" it, we'd need to move it to the pages folder for consistency.

The end user of the site wouldn't see that workflow but people who would have consulted the actual wip-tec documents would lose the access to that URL at the time which isn't right.

**Comment by gcharest** [comment]
@smellems if you have a few cycles in the upcoming week, setting up HTMLProofer and HTML5Validator could be a great addition to our tests.

If not, I'll try to have a look myself.

**Comment by smellems** [comment]
#278 enables the markdown link checking for external links.

**Comment by smellems** [comment]
Markdown link checking is now working during the build.  This checks links in markdown files and not the resulting HTML. but It's good to validate that external link work.  Internal links are now shown as warnings so that we can see what they are and fix them as needed.

This test can be run using `npm run link-check`

The HTML on the live site can be validate using `npm run site-validator`

This will crawl internal links and validate the HTML but I don't think it fails on dead links
